sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryAsyncClient.java

```diff
@@ -0,0 +1,186 @@
+ */
+@ServiceClient(builder = ContainerRegistryClientBuilder.class, isAsync = true)
+public final class ContainerRegistryAsyncClient {
+    private final ContainerRegistriesImpl serviceClient;
+    private final ClientLogger logger = new ClientLogger(ContainerRegistryAsyncClient.class);
+    private static final String CONTINUATIONLINKHEADERNAME = "Link";
+    private static final String LINKMATCHER = "(<(.+)>;.*)";
+    private static final String ORDERBYQUERYPARAM = "&orderby=";
+
+    /**
+     * Initializes an instance of ContainerRegistries client.
+     *
+     * @param serviceClient the service client implementation.
+     */
+    ContainerRegistryAsyncClient(ContainerRegistriesImpl serviceClient) {
+        this.serviceClient = serviceClient;
+    }
+
+    /**
+     * List repositories.
+     *
+     * @return list of repositories.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws AcrErrorsException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<String> listRepositories() {
+        ListRepositoriesOptions options = new ListRepositoriesOptions();
+        return this.listRepositories(options);
+    }
+
+    /**
+     * List repositories.
+     *
+     * @param options for the list repositories call.
+     * @return list of repositories.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws NullPointerException thrown if the parameter is null.
+     * @throws AcrErrorsException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<String> listRepositories(ListRepositoriesOptions options) {
+        if (options == null) {
+            throw logger.logExceptionAsError(new NullPointerException("options can't be null"));
+        }
+
+        Integer pageSize = options.getMaxPageSize();
+        return new PagedFlux<>(
+            () -> withContext(context -> listRepositoriesSinglePageAsync(pageSize, context)),
+            token -> withContext(context -> listRepositoriesNextSinglePageAsync(token, context)));
+    }
+
+    private PagedResponse<String> getPagedResponseWithContinuationToken(PagedResponse<String> listRepositoriesResponse) {
+        String continuationLink = null;
+        HttpHeaders headers = listRepositoriesResponse.getHeaders();
+
+        if (headers != null) {
+            String continuationLinkHeader = headers.getValue(CONTINUATIONLINKHEADERNAME);
+            if (!CoreUtils.isNullOrEmpty(continuationLinkHeader)) {
+                // TODO: ACR follows docker API spec which has continuation token in header.
```

Comment:
nit: TODOs should use the pattern `TODO (<alias>)`

Also mind writing an issue to follow-up on this.

Reply:
Already done. They are working on putting the link in the response. Also, we have a work-item on our side to parse the link header along with the continuationToken in the response.

---
In reply to: [602630116](https://github.com/Azure/azure-sdk-for-java/pull/20139#discussion_r602630116) [](ancestors = 602630116)